packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -326,4 +326,101 @@ export class MultichainAccountWallet<
       await group.align();
     }
   }
+
+  /**
+   * Discover and create accounts for all providers.
+   *
+   * NOTE: This method should only be called on a newly created wallet.
+   *
+   * @param opts - The options for the discovery and creation of accounts.
+   * @param opts.skippedProviders - The providers to skip.
+   * @returns The accounts for each provider.
+   */
+  async discoverAndCreateAccounts({
+    skippedProviders = [],
+  }: {
+    skippedProviders: AccountProviderType[];
```

Comment:
I feel like having this forces us to have an `AccountProviderType`, which I believe we don't need it because:
- It prevents custom account providers (since they cannot extend the `enum`)
- We don't really need to identify any of the provider for now in the system

That being said, that could be useful for debugging purposes, but I'd rather prefer having a `string` identifier which could resolve to anything that makes sense for the provider (`snapId` for Snap-based providers, the `keyring.type` for keyring-based providers for example).

WDYT?